### ISSUE TITLE
Unified Hyphen in [Hexadecimal numeric character references]

### ISSUE THREAD

**Issue by KiyanYang** [commonmark/commonmark-spec#728]
https://github.com/commonmark/commonmark-spec/blob/796666de5398558be184059c8b00077822f13662/spec.txt#L676-L680

The hyphen in paragraph about  Hexadecimal numeric character references, which different from other places. So I suggest to change `a string of 1-6 hexadecimal digits` into `a string of 1--6 hexadecimal digits`. Here are examples of other locations:

https://github.com/commonmark/commonmark-spec/blob/796666de5398558be184059c8b00077822f13662/spec.txt#L661-L667

https://github.com/commonmark/commonmark-spec/blob/796666de5398558be184059c8b00077822f13662/spec.txt#L1099-L1109

https://github.com/commonmark/commonmark-spec/blob/796666de5398558be184059c8b00077822f13662/spec.txt#L2993-L2994

https://github.com/commonmark/commonmark-spec/blob/796666de5398558be184059c8b00077822f13662/spec.txt#L4106-L4110

**Comment by wooorm** [comment]
I’d personally prefer to use unicode characters in the spec, rather than ancient ASCII-era typewriter pseudocharacters 😅

**Comment by jgm** [comment]
Yes, it should indeed be an `--` which will turn into an en dash in the rendered version.
